### DISCUSSION TITLE
How to pass custom data to table builder

### DISCUSSION THREAD

**Discussion by romaldyminaya** [filamentphp/filament#1610] [Help] [answered]
Hi,

I would like to use the table builder a a report, passing custom data that isn't associated with eloquent models.

Is that possible ?

**Comment by danharrin** [comment, marked as answer]
You always have to use Eloquent I'm afraid. Feel free to use the Blade components within `/tables/resources/views/components` to make your table look the same though.

**Comment by ceejayoz** [comment]
Just a late note that you can use the Sushi package's `getRows()` functionality for this. It'll let you make a pseudo-model that can get its backing data from essentially anywhere; external API, union of several tables, etc.

https://filamentphp.com/community/how-to-consume-an-external-api-with-filament-tables

**Reply by gru2007** [reply]
Link has been updated 
https://filamentphp.com/content/leandrocfe-how-to-consume-an-external-api-with-filament-tables

I need to get data from Redis, so, good luck to me :)

**Comment by jeremyferg23** [comment]
We had a similar use case with an external api. Sushi trait basically hijacks Eloquent but your return type must be an array in an array. there's no need to use the `getRows()` anymore.

We are using a laravel get attribute on the model for the api `Product::find(1)->apiReport->toArray();` and returns

`[
    [ "product_name" => "Whole Chicken" ],
    [ "product_name" => "Half Chicken" ]
]`
